rpcs3/Emu/Cell/lv2/lv2.cpp

```diff
@@ -1040,21 +1040,25 @@ void lv2_obj::sleep_timeout(cpu_thread& thread, u64 timeout)
 		ppu->start_time = start_time;
 	}
 
-	if (timeout && g_cfg.core.sleep_timers_accuracy != sleep_timers_accuracy_level::_all_timers)
+	if (timeout)
 	{
 		const u64 wait_until = start_time + timeout;
 
 		// Register timeout if necessary
-		for (auto it = g_waiting.begin(), end = g_waiting.end(); it != end; it++)
+		for (auto it = g_waiting.begin(), end = g_waiting.end();; it++)
 		{
+			if (it == end)
+			{
+				g_waiting.emplace_back(wait_until, &thread);
+				break;
+			}
+
 			if (it->first > wait_until)
```

Comment:
Can write as `it == end || it->first > wait_until`

Reply:
emplace_back() is more optimized than emplace() because it doesn't need to rotate other elements contents when emplacing and perform checks related to it, so if we know we are going to emplace at end(), better to use emplace_back() to avoid redundant checks.